ocs_ci/ocs/platform_nodes.py

```diff
@@ -818,6 +824,271 @@ def _prepare_node(self, node_id):
+        )
+
+        delete_file(worker_template_path)
+        return self.aws.get_ec2_instance(instance_id)
+
+    def create_stack(self, conf, s3_url):
+        """
+        Create a new cloudformation stack for worker creation
+
+        Args:
+            conf (dict): of worker configuration
+            s3_url (str): An aws url for accessing s3 object
+        """
+        index = conf['node_id']
+        stack_name = f"{self.cluster_name}-no{index}"
+
+        try:
+            response = self.cf.create_stack(
+                StackName=stack_name,
+                TemplateURL=s3_url,
+                Parameters=self.build_stack_params(index, conf),
+                Capabilities='CAPABILITY_NAMED_IAM'
+            )
+        except Exception:
+            logger.error("Cloudformation Stack creation failed")
+            raise
+        self.cf.get_waiter('stack_create_complete').wait(StackName=stack_name)
+        logger.info(f"Stack {stack_name} created successfuly")
+        self.stack_name = stack_name
+        self.stack_id = response['StackId']
+        logger.info(f"Stackid = {self.stack_id}")
+
+    def build_stack_params(self, index, conf):
+        """
+        Build all the params required for a stack creation
+
+        Args:
+            index (int): An integer index for this stack
+            conf (dict): Node config
+
+        Returns:
+            list: of param dicts
+
+        """
+        param_list = []
+        pk = 'ParameterKey'
+        pv = 'ParameterValue'
+
+        param_list.append({pk: 'Index', pv: index})
+        param_list.append({pk: 'InfrastructureName', pv: self.infra_id})
+        param_list.append({pk: 'RhcosAmi', pv: self.worker_image_id})
+        param_list.append(
+            {
+                pk: 'IgnitionLocation', pv: self.worker_ignition_location
+            }
+        )
+        param_list.append({pk: 'Subnet', pv: self.worker_subnet})
+        param_list.append(
+            {
+                pk: 'WorkerSecurityGroupId', pv: self.worker_security_group
+            }
+        )
+        param_list.append(
+            {
+                pk: 'WorkerInstanceProfileName', pv: self.worker_instance_profile
+            }
+        )
+        param_list.append(
+            {
+                pk: 'WorkerInstanceType', pv: conf['worker_instance_type']
+            }
+        )
+
+        return param_list
+
+    def add_cert_to_template(self, worker_template_path):
+        """
+        Add cert to worker template
+
+        Args:
+            worker_template_path (str): Path where template file is located
+
+        """
+        worker_ignition_path = os.path.join(
+            self.cluster_path,
+            constants.WORKER_IGN
+        )
+        cert = self.get_cert_content(worker_ignition_path)
+        self.update_template_with_cert(worker_template_path, cert)
+
+    @retry(exceptions.PendingCSRException, tries=4, delay=10, backoff=1)
+    def approve_pending_csr(self):
```

Comment:
Could you move approve_pending_csr to utility so that it can be used for both AWS and vSphere

Reply:
Coady is moving this to be independent function, once that PR is merged I will remove these functions from here.